README.md

```diff
@@ -1,168 +1,156 @@
+  Crucially, prerendering browsing contexts have the ability to transition to becoming normal top-level browsing contexts, so that all of the prerendered content is reused and immediately displayed to the user.
 
-## Request header
+* [**Portals**](https://github.com/WICG/portals/blob/master/README.md), which are a specialization of prerendering browsing contexts which can display a preview of the prerendered content to the user, and which expose a JavaScript API for transitioning to a normal top-level browsing context.
 
-If an alternate loading mode is used, the user agent will send a `Sec-Loading-Mode` request header indicating the mode. The server may respond with an error status code (400-599), in which case prefetching/prerendering will be considered to have failed, regardless of the response body.
+Each of these pieces is connected in various ways. However, we think they're decoupled enough that we will start by developing them as separate explainers and spec documents, cross-linking to each other as appropriate. It's also possible to implement only a subset of these, if a user agent is only interested in certain [scenarios](#example-scenarios).
 
-A server could defer non-idempotent behavior such as impression counting,  or it could reject prerender requests altogether without needing to do the full work to generate a response (since it need only inspect the request headers).
+## Same-origin vs. cross-origin prerendering
 
-> TODO: Should we also/instead send `Sec-Purpose`? Should `Sec-Fetch-Mode` differ?
+When prerendering same-origin content, many fewer constraints are necessary. Because there is no privacy concern, we can use normal fetching modes, and thus do not need an opt-in from the prerendered page. And the prerendering browsing context becomes simpler. Thus, the majority of work for specifying same-origin prerendering is in the prendering triggers, the restrictions on disruptive APIs, and the transition to a normal top-level browsing context.
 
-## Risks
+This simplicity benefits web developers as well, as they don't need to do the upgrade-from-uncredentialed dance which is necessary in cross-origin cases.
 
-__Credentialed requests__ allow fetching the right resource for the user, but the web is increasingly moving toward a model whereby an origin cannot trigger credentialed requests to another user except as part of a committed action (i.e., a navigation). Forward-looking prerendering should prefer to use uncredentialed requests to reduce the risk of identifying the user when doing so is undesirable, and should also deny access to unpartitioned storage under those circumstances.
+The tradeoff is that we now require opt-in from the referring page. The user agent cannot just heuristically prefetch or prerender any same-origin links that it sees; doing so would have bad consequences for links like `<a href="/logout">`.
```

Comment:
I don't quite understand why opt-in is required for same-origin. If there's no opt-in, could it make sense to do automatic prerendering for a same-origin link and just use the same restrictions as cross-origin prerendering?

I guess we assume if the site bothered to put opt-in on the content response, it would have bothered to put opt-in on the referring page if it were worthwhile, so there's no need to try automatic prerendering?

Reply:
I think you're right; we could do same-origin prerendering with only destination-side opt-in. (Would we also omit credentials? Hmm.)

Since this section is supposed to be comprehensive, I think I should mention that, even if we end up deciding to build something that sticks to the simpler rule I propose here. (I.e., sticks to triggering-page opt-in for same-origin, destination-page opt-in for cross-origin.)

Reply:
Destination side opt-in only is a problem because you don't know if the GET will have side effects, which is especially acute in the credentialed (same-origin) case.

Reply:
Why would a side-effecting page opt-in?

Reply:
You cannot tell if a destination page has opted in until after you have already fetched it.

Reply:
So we could do same-origin destination-side only opt-in as long as we applied the same restrictions that we do for cross-origin, i.e. no credentials/storage access/etc. Right? I'll push a commit mentioning that possibility, but saying that we're not currently pursuing it.

Reply:
I'm not sure if mentioning same-origin destination-side opt-in case is important or necessary at this point, it may just confuse readers?  One thing that's clear is that we'll start with referrer side API (either in implicit or explicit)

Reply:
Ah I saw the new text below. I'm good with the current framing.

Reply:
Right, but I'd prefer to err on the side of not sending a GET unless we have *some* sort of indication that it's not going to have side effects. Being uncredentialed lowers the bar, but I'm not sure I'm ready to lower the bar to zero.